File: client/src/app/_models/products/productParams.ts
import { EntityParams } from "src/app/_models/base/entityParams";
import { Product } from "src/app/_models/products/product";
import { SelectOption } from "src/app/_models/base/selectOption";


export class ProductParams extends EntityParams<Product> {
  isInternal: boolean | null = null;
  sku: string | null = null;
  barcode: string | null = null;
  lotNumber: string | null = null;
  price: number | null = null;
  category: SelectOption[] | null = null;
  manufacturer: SelectOption[] | null = null;

  constructor(key: string | null, init?: Partial<ProductParams>) {
    super(key);
    Object.assign(this, init);
  }
}

File: src/MainService.Core/Helpers/Params/ProductParams.cs
using MainService.Models.Entities.Aggregate;

namespace MainService.Core.Helpers.Params;

public class ProductParams : BaseParams
{
    public bool? IsInternal { get; set; }
    public int? DoctorId { get; set; }
    public string? Barcode { get; set; }
    public string? Sku { get; set; }
    public int? Price { get; set; }
    public string? LotNumber { get; set; }
    public List<OptionDto>? Category { get; set; }
    public List<OptionDto>? Manufacturer { get; set; }
}